### PR TITLE
 slogdet #2462 

### PR DESCRIPTION
**Pull request by MohamedHaggag-Cloud**

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -6,3 +6,9 @@ def det(input, name=None):
 
 
 det.unsupported_dtypes = ("float16", "bfloat16")
+
+
+def slogdet(input, name=None):
+    return ivy.slogdet(input)
+
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/tensorflow/linalg.py:13]: Please add a blank line here to fix the lint failure.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -6,3 +6,9 @@ def det(input, name=None):
...
+slogdet.unsupported_dtypes = ("float16", "bfloat16")
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/tensorflow/linalg.py:14]: This should be converted to a dictionary indicating the backend which doesn't support that particular dtype for your function, the [docs](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html) have more details regarding the same.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 # local
 import ivy_tests.test_ivy.helpers as helpers
 import ivy.functional.backends.tensorflow as ivy_tf
+import ivy.functional.backends.numpy as ivy_np
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:8]: We should be using `ivy_tf` for generating data for tensorflow frontend tests

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 4]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
         fn_tree="linalg.det",
         input=np.asarray(x, dtype=input_dtype),
     )
+
+
+# slogdet
+@handle_cmd_line_args
+@given(
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes,
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:56]: Please update your test to use `ivy_tf.valid_float_dtypes` instead.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 5]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
+        min_value=0,
+        max_value=50,
+        shape=helpers.ints(min_value=2, max_value=20).map(lambda x: tuple([x, x])),
+    ),
+    num_positional_args=helpers.num_positional_args(fn_name="slogdet"),
+)
+def test_slogdet(
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:63]: The function name should be `test_tensorflow_slogdet`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 6]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
         fn_tree="linalg.det",
         input=np.asarray(x, dtype=input_dtype),
     )
+
+
+# slogdet
+@handle_cmd_line_args
+@given(
+    dtype_x=helpers.dtype_and_values(
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:55]: please update this name to `dtype_and_x` as it's a commonly used name across the codebase for tests.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 7]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
+        max_value=50,
+        shape=helpers.ints(min_value=2, max_value=20).map(lambda x: tuple([x, x])),
+    ),
+    num_positional_args=helpers.num_positional_args(fn_name="slogdet"),
+)
+def test_slogdet(
+    *,
+    dtype_x,
+    as_variable,
+    with_out,
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:67]: As the function doesn't support an `out` argument, no need for this argument in the function declaration.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 8]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
+    helpers.test_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        container_flags=container,
+        instance_method=instance_method,
+        fw=fw,
+        fn_name="slogdet",
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:84]: Please also pass a value to the `frontend` argument.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 9]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    input_dtype, x = dtype_x
+    helpers.test_function(
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:75]: The helper function used should be `test_frontend_function`.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 8 above)
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:84]: The argument name `fn_name` should be renamed to `fn_tree`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 10]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
     )
+
+
+# slogdet
+@handle_cmd_line_args
+@given(
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes,
+        min_value=0,
+        max_value=50,
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:58]: Please remove the max_value limit, as we should be testing the function exhaustively.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 11]
```diff
@@ -45,3 +47,40 @@ def test_tensorflow_det(
...
+        shape=helpers.ints(min_value=2, max_value=20).map(lambda x: tuple([x, x])),
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:59]: Please update `max_value` to have 10 instead, as it's the default value of `max_dim_size` in the `dtype_and_values` helper.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @MohamedHaggag-Cloud, thanks for contributing! There are a few changes to be made to your PR before we get it merged. I have requested changes near the lines of code concerned, feel free to make those changes and re-request my review once your done, after which we'll take a look at the results of the tests.
Let me know if you have any questions. Thanks 🙂

**Comment by VedPatwardhan** [comment]
Hi @MohamedHaggag-Cloud, please let me know if you're working on this PR, because otherwise, it may have to be closed in a couple of days due to inactivity. Thanks 🙂

**Comment by MohamedHaggag-Cloud** [comment]
Hi, I am working on it, but I tried to pull the newest pull, then the pull request closed by itself. But I am working on your modifications

File: ivy/functional/frontends/tensorflow/linalg.py (hunk 1 above)
Comment (MohamedHaggag-Cloud) [reply ivy/functional/frontends/tensorflow/linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 4 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 5 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy/functional/frontends/tensorflow/linalg.py (hunk 2 above)
Comment (MohamedHaggag-Cloud) [reply ivy/functional/frontends/tensorflow/linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 3 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 7 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 8 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 9 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 8 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 10 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 11 above)
Comment (MohamedHaggag-Cloud) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: done

**Pull request merged**